src/components/App/App.js

```diff
@@ -40,6 +43,33 @@ class App extends Component {
     })
   }
 
+  setFavorites = (newFavorite) => {
+    const allIDs = this.state.favorites.reduce((acc, favorite) => {
+      acc.push(favorite.id);
+      return acc;
+    }, []);
+    if (!allIDs.includes(newFavorite.id)) {
+      this.saveFavorite(newFavorite);
+    } else {
+      this.removeFromSaved(newFavorite);
+    }
+  };
+
+  saveFavorite = (favorite) => {
+    this.setState({
+      favorites: [...this.state.favorites, favorite],
+    });
+  };
+
+  removeFromSaved = (favoriteToRemove) => {
+    const newFavorites = this.state.favorites.filter((favorite) => {
+      return favorite.id !== favoriteToRemove.id;
+    });
+    this.setState({
+      favorites: newFavorites,
+    });
+  };
```

Comment:
Called by above method - depending on the path, one will be called, to either remove or add the favorite.